src/twister2/device/hardware_adapter.py

```diff
@@ -145,7 +147,9 @@ def flash_and_run(self, timeout: float = 60.0) -> None:
                 for line in stdout.decode('utf-8').split('\n'):
                     if line:
                         logger.info(line)
-
+            finally:
```

Comment:
Code in `finally` will be always executed, but stdout and stderr will have some value only if there wasn't any exception, so it make sense to run this code:
```python 
if stderr:
   self.device_log_file.handle(data=stderr)
```
only if there was not an exception, so `finally` should  be replaced by `else` and line 141 can be removed.

Reply:
Done